### ISSUE TITLE
Tests using file download don't detect new bugs

### ISSUE THREAD

**Issue by DamienCassou** [cypress-io/cypress#14886]
### Current behavior

1. you have a test checking that a download button works properly;
2. you open Cypress with `cypress open`;
2. the test passes and a file is saved in `cypress/downloads/`;
3. you change your code and introduce a bug preventing the download button from working;
4. you restart the test in Cypress

Expected: the test fails because the code of the download button has a bug
Actual: the test passes because the file downloaded previously still exists

I would expect the files in `cypress/downloads/` to be deleted before each test run automatically. Alternatively, a less satisfying solution would introduce a `cy.cleanDownloadsFolder()` command that I would call manually in my `beforeEach()` hooks.

This problem happens because `trashAssetsBeforeRuns` is only taken care of for `cypress run`, not for `cypress open`. This [confuses users](https://github.com/cypress-io/cypress/issues/14870).

### Versions

Cypress 6.3.0.

**Comment by jennifer-shehane** [comment]
Yeah, we actually encountered this situation ourselves. Mentioned in this thread where we didn't catch the downloads not working: https://github.com/cypress-io/cypress-example-recipes/pull/631

>The false positives highlight the danger of only clearing downloads between runs and not between tests. If a user tests the same file in multiple tests, which is a fairly likely chance, there's a risk of such a false positive. We should think about changing it so downloads are cleared between tests (with a config option to disable).

**Comment by RodeoMike** [comment]
I'd propose not clearing the downloads. I have scripts that perform applying various filters and then downloads a report based on the filters. In the next test (within the same script), different filters are applied and then the report is downloaded again. What's happening now is the report is simply overwritten, preventing me from manually validating the results in the reports. Prior to the download feature being added in 6.3.0, each report would append (#) to the file name.  

If this needs a new bug, I can certainly open one.

**Comment by jennifer-shehane** [comment]
@RodeoMike It'd be really helpful to see an example of your test code, even psuedo code would be fine, because there was some debate over whether people would need to access the downloads outside of the current test - and real world use cases are helpful here. Thanks.

**Comment by RodeoMike** [comment]
@jennifer-shehane Sure! I've only been using Cypress for a few months and basically just winging it, but this is one example.

  it('4-Filter on Barn 11. Downloads report, then clears filter', () => {
    // Open filter
    cy.get('[data-testid="filters-toggle"]').click()
    cy.get('[id="mui-component-select-STALL LOCATION"]').click()
    // Get the Barn 11 filter
    cy.contains('Barn 11').click()

    // Apply filter
    cy.get('[data-testid="filters-apply-btn"]').click()
    cy.contains('EXPORT REPORT').click()
    cy.wait(1000) //wait 1 second
    cy.get('[class="MuiTable-root"]').contains('CLEAN')
    
    I have additional reports using different logins that require I manually check calculations too (the above example is just a list). So in order to test those I have to execute the script manually so that the reports don't get over-written.
